File: sportstream-dashboard/src/types/club.ts
export interface Club {
  id: string;
  name: string;
  slug: string;
  country: string;
  league: string;
  logoUrl: string;
  sport: string;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
}

export type CreateClubPayload = Omit<Club, 'id' | 'createdAt' | 'updatedAt'>;
export type UpdateClubPayload = Partial<CreateClubPayload>;

File: sportstream-api/internal/domain/club.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type Club struct {
	ID        uuid.UUID `json:"id"`
	Name      string    `json:"name"`
	Slug      string    `json:"slug"`
	Country   string    `json:"country"`
	League    string    `json:"league"`
	LogoURL   string    `json:"logo_url"`
	Sport     string    `json:"sport"`
	IsActive  bool      `json:"is_active"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

type ClubRepository interface {
	FindAll() ([]Club, error)
	FindByID(id uuid.UUID) (*Club, error)
	Create(club *Club) error
	Update(club *Club) error
	Delete(id uuid.UUID) error
}
